### PR TITLE
Add skip_cleared option to error_count attribute to not serialise errors that have never been initialised

### PR DESCRIPTION
**Pull request by w4**
This option defaults to `false` so this is a non-breaking change and the original functionality still stands. If this option is set to `true`, the error will not be serialised out to the output if it has never been returned from the function being measured.

### CODE REVIEW

File: metered/src/metric.rs [hunk 1]
```diff
@@ -61,6 +61,9 @@ impl<'a, R, M: Metric<R>> Drop for ExitGuard<'a, R, M> {
 pub trait Counter: Default + Clear + Serialize {
     /// Increment the counter
     fn incr(&self);
+
+    /// Returns true if the counter is set to 0.
+    fn is_cleared(&self) -> bool;
```
Comment (magnet) [inline metered/src/metric.rs:66]: Should this be defined in the `Clear` trait, or even a separate trait?  I see the problem with implementing this for Histograms: 0 can be a measured value on some histograms.

Comment (w4) [reply metered/src/metric.rs]: That makes sense, I've taken this out into a separate `Clearable` trait.

**Comment by magnet** [comment]
I've renamed the feature flag to `error-count-skip-cleared-by-default` and documented it.

**Comment by magnet** [comment]
Pushed to `master` manually.

**Pull request closed without merging**